client/pluginManger.go

```diff
@@ -104,20 +106,24 @@ func (p *PluginManager) GetPlugins() (pluginList *InstalledPluginList, err error
 	return
 }
 
-func getPluginsInstallQuery(names []string) string {
+func (p *PluginManager) getPluginsInstallQuery(names []string) string {
 	pluginNames := make([]string, 0)
 	for _, name := range names {
 		if name == "" {
 			continue
 		}
+		if strings.Contains(name, "@") {
+			p.installPluginWithVersion(name)
+			continue
+		}
 		pluginNames = append(pluginNames, fmt.Sprintf("plugin.%s=", name))
 	}
 	return strings.Join(pluginNames, "&")
 }
 
 // InstallPlugin install a plugin by name
 func (p *PluginManager) InstallPlugin(names []string) (err error) {
-	api := fmt.Sprintf("/pluginManager/install?%s", getPluginsInstallQuery(names))
+	api := fmt.Sprintf("/pluginManager/install?%s", p.getPluginsInstallQuery(names))
```

Comment:
Maybe you can create a function named `getVersionalPlugins` here to install and upload them.